apps/examples/slsiwifi/slsiwifi_main.c

```diff
@@ -164,6 +170,23 @@ void sw_linkUpHandler(slsi_reason_t *reason)
 		printf_encode(connectedApName, WPA_MAX_SSID_LEN, reason->ssid, reason->ssid_len);
 		printf("Connected to network: bssid: %s, ssid: %s\n", reason->bssid, connectedApName);
 		sem_post(&g_sem_join);	//tell the main thread to move on
+
+		/*
+		 Start DHCP client procedures after STA is connected to AP
+		 Aftrer successful DHCP, IP address will be assigned to the WLAN Interface
+		*/
+		dhcp_handle = dhcpc_open(WLAN_INTF_NAME);
+		if (dhcp_handle == NULL) {
+			dmdbg("Invalid dhcp handle\n");
+			return;
+		}
+		if (dhcpc_request(dhcp_handle, &state) != OK) {
+			dhcpc_close(dhcp_handle);
+			return;
+		}
+		netlib_set_ipv4addr(WLAN_INTF_NAME, &state.ipaddr);
+		netlib_set_ipv4netmask(WLAN_INTF_NAME, &state.netmask);
+		netlib_set_dripv4addr(WLAN_INTF_NAME, &state.default_router);
 	} else {
```

Comment:
This setting is static for ipv4. Restricting it to only ipv4 is not good idea. Can the test case be made compatible to work for ipv4 and v6 both. You may have to provide another wrapper (or may be already exist)